config/confighttp/compression_test.go

```diff
@@ -104,10 +126,17 @@ func TestHTTPClientCompression(t *testing.T) {
 
 			req, err := http.NewRequest(http.MethodGet, srv.URL, reqBody)
 			require.NoError(t, err, "failed to create request to test handler")
-
 			clientSettings := ClientConfig{
-				Endpoint:    srv.URL,
-				Compression: tt.encoding,
+				Endpoint:          srv.URL,
+				Compression:       tt.encoding,
+				CompressionParams: newCompressionParams(tt.level),
+			}
+			err = clientSettings.Validate()
+			if tt.shouldError {
+				require.Error(t, err)
+				message := fmt.Sprintf("unsupported compression type and level %s - %d", tt.encoding, tt.level)
+				assert.Equal(t, message, err.Error())
+				return
 			}
```

Comment:
I can't seem to unresolve comments (https://github.com/open-telemetry/opentelemetry-collector/pull/11805/files#r1904797587), so creating a new one: should there be a `require.NoError(t, err)` in the `tt.shouldError == false` branch?

Reply:
yes added that after the `if` condition.